app/src/main/java/com/github/libretube/ui/controllers/FullscreenGestureAnimationController.kt

```diff
@@ -0,0 +1,96 @@
+package com.github.libretube.ui.controllers
+
+import android.view.MotionEvent
+import android.view.animation.AccelerateDecelerateInterpolator
+import com.github.libretube.extensions.dpToPx
+import com.github.libretube.ui.views.CustomExoPlayerView
+
+/**
+ * A class that handles the scale animation in [CustomExoPlayerView] when entering/exiting
+ * fullscreen using the swipe gesture.
+ */
+class FullscreenGestureAnimationController(
+    private val playerView: CustomExoPlayerView,
+    private val onSwipeUpCompleted: () -> Unit,
+    private val onSwipeDownCompleted: () -> Unit,
+) {
+    private var isSwipeInProgress = false
+    private var shouldHandleSwipe = false
+    /**
+     * - `value > 0` = Swiping up
+     * - `value < 0` = Swiping down
+     */
+    private var swipeDirection = 0
```

Comment:
Since you're not using the actual value of `swipeDirection`, this could also be an enum type with the values `up, none, down` for better readability